docs/index.rst

```diff
@@ -27,123 +27,166 @@ EdgeDB TypeScript/JS Client
    group
    reference
 
-This is the official EdgeDB client library for JavaScript and TypeScript. It's
-the easiest way to connect to your database and execute queries from a Node.js
-or Deno backend.
-
-.. note::
-
-  We recently released ``v1.0.0`` of the ``edgedb`` module on NPM. This is a major version release and comes with breaking changes and major new features. Notably, the workflow for generating the query builder has changed significantly. Refer to the `release notes <https://github.com/edgedb/edgedb-js/releases>`_ for details.
-
-There are two components of this library:
-
-- Use the :ref:`Client API <edgedb-js-intro-driver>` to establish a connection
-  to your database and execute EdgeQL queries (written as strings).
-- Use the :ref:`Query Builder API <edgedb-js-intro-qb>` to write queries in a
-  code-first, typesafe way. Recommended for TypeScript users.
-
 .. _edgedb-js-installation:
 
-**Requirements**
-
-*Node.js*
-  Node.js 14+. Run ``node --version`` to see your current version.
-
-*Deno*
-  Deno v1.17+. Run ``deno --version`` to see your current version.
-
-*TypeScript*
-  - TypeScript v4.4+
-  - Node.js type declarations: ``npm install @types/node``
-  - Compiler options: ``"strict": true`` and ``"downlevelIteration": true`` in
-    ``tsconfig.json``
 
-**Installation**
+Installation
+============
 
-*Node users*: Install the ``edgedb`` module from NPM.
+The database driver and optional (but recommended!) generators are published to
+npm and deno.land, and can be installed with your package manager of choice.
 
-.. code-block:: bash
+.. tabs::
 
-    $ npm install edgedb      # npm users
-    $ yarn add edgedb         # yarn users
+    .. code-tab:: bash
+      :caption: npm
 
-*Deno users*: Import from ``deno.land/x/edgedb``:
+      $ npm install --save-prod edgedb          # database driver
+      $ npm install --save-dev @edgedb/generate # generators
 
-.. code-block:: typescript
+    .. code-tab:: bash
+      :caption: yarn
 
-  import * as edgedb from "https://deno.land/x/edgedb/mod.ts"
+      $ yarn add edgedb                 # database driver
+      $ yarn add --dev @edgedb/generate # generators
 
-.. _edgedb-js-intro-driver:
+    .. code-tab:: bash
+      :caption: pnpm
 
-The ``Client`` API
-==================
+      $ pnpm add --save-prod edgedb          # database driver
+      $ pnpm add --save-dev @edgedb/generate # generators
 
-The ``Client`` class implements the core functionality required to establish a
-connection to your database and execute queries. If you prefer writing queries
-as strings, the Client API is all you need.
+    .. code-tab:: typescript
+      :caption: deno
 
-.. code-block:: javascript
+      import * as edgedb from "http://deno.land/x/edgedb/mod.ts";
 
-  const edgedb = require("edgedb");
+Requirements
```

Comment:
I think it makes sense to keep these above installation since that's how it works chronologically — meet the requirements, then install.

Reply:
Yeah, that's fair, I was definitely optimizing for visual scanning where you land on this page and just want to install it and the requirements bit is something you might not ever look at since you'd probably only notice you don't meet the requirements if something isn't working.

Also, what do you think about how I've stated the requirements? Basically instead of saying minimum versions (which is hard to keep in mind while adding features), I've listed what versions we actually test that they work?

Reply:
That makes sense. I think you've shortened the requirements enough that you don't lose much by bumping it up.

I like that approach! I might flip the perspective to instead focus on our user instead of us. I'll write a proper suggestion below.

Reply:
Actually, now I'm second guessing that. 😅  I would generally prefer to write from the user perspective where it's currently written from our perspective: "We test…" vs. "You should use…" (or something similar)

But actually I think what you're trying to do is to say that it may work under some other version. By saying "We test…" you've leaving that possibility open. Am I understanding your intent?

Reply:
Yeah, that's my intent: these are not minimum versions, they are the versions we promise to test and support. They may work on earlier versions, and there are probably currently known versions, but without testing on those versions there is no way for us to know if we've broken capability with those versions.

Reply:
To be honest, I could see just moving this whole section to the repo's README since it probably introduces more friction than it alleviates. Our support posture isn't that radical here, and it's not likely to save us from anything unless we have some kind of runtime/install-time checking in place.

Reply:
That sounds fine to me, @scotttrinh!